File: src/LCR024.UHnkqh.1000260/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode dummyHead = new ListNode(0);

        while (head != null){
            ListNode next = head.next;
            head.next = dummyHead.next;
            dummyHead.next = head;
            head = next;
        }
        return dummyHead.next;
    }
}

File: src/LCR024.UHnkqh.1000260/solution.go
package offer100260

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
	dummyHead := ListNode{}
	for head != nil {
		n := head.Next
		head.Next = dummyHead.Next
		dummyHead.Next = head
		head = n
	}
	return dummyHead.Next
}
